internal/attestation/renderer/renderer.go

```diff
@@ -87,15 +106,85 @@ func (ab *AttestationRenderer) Render() (*dsse.Envelope, error) {
 		return nil, err
 	}
 
-	signedAtt, err := ab.signer.SignMessage(bytes.NewReader(rawStatement))
+	signedAtt, err := ab.dsseSigner.SignMessage(bytes.NewReader(rawStatement))
 	if err != nil {
 		return nil, fmt.Errorf("signing message: %w", err)
 	}
 
-	var dseeEnvelope dsse.Envelope
-	if err := json.Unmarshal(signedAtt, &dseeEnvelope); err != nil {
+	var dsseEnvelope dsse.Envelope
+	if err := json.Unmarshal(signedAtt, &dsseEnvelope); err != nil {
 		return nil, err
 	}
 
-	return &dseeEnvelope, nil
+	if ab.bundlePath != "" {
+		// Create sigstore bundle for the contents of this attestation
+		bundle, err := ab.envelopeToBundle(dsseEnvelope)
+		if err != nil {
+			return nil, fmt.Errorf("loading bundle: %w", err)
+		}
+		json, err := protojson.Marshal(bundle)
+		if err != nil {
+			return nil, fmt.Errorf("marshalling bundle: %w", err)
+		}
+		ab.logger.Info().Msg(fmt.Sprintf("generating Sigstore bundle %s", ab.bundlePath))
+		err = os.WriteFile(ab.bundlePath, json, 0600)
+		if err != nil {
+			return nil, fmt.Errorf("writing bundle: %w", err)
+		}
+	}
+
+	return &dsseEnvelope, nil
+}
+
+func (ab *AttestationRenderer) envelopeToBundle(dsseEnvelope dsse.Envelope) (*protobundle.Bundle, error) {
```

Comment:
It would be great to add some unit tests on this. I believe somewhere else I did some `golden` tests that rely on `snapshots` of the output. In other words, you can put envelopes as inputs and bundles as outputs, and the test just check those.